paymentsheet/src/main/java/com/stripe/android/paymentsheet/Element.kt

```diff
@@ -76,11 +78,28 @@ internal sealed class FormElement {
 
     data class SectionElement(
         override val identifier: IdentifierSpec,
-        val field: SectionFieldElementType,
-        override val controller: InputController
-    ) : FormElement(), OptionalElement
+        val fields: List<SectionFieldElementType>,
+        override val controller: SectionController
+    ) : FormElement(), OptionalElement {
+        internal constructor(
+            identifier: IdentifierSpec,
+            field: SectionFieldElementType,
+            controller: SectionController
+        ) : this(identifier, listOf(field), controller)
+    }
 }
 
+internal fun List<FormElement>.idInputControllerMap() = this
+    .filter { it.controller is InputController && it.controller != null }
+    .associate { it.identifier to (it.controller as InputController) }
+    .plus(
+        this
+            .filterIsInstance<FormElement.SectionElement>()
+            .map { it.fields }
+            .flatten()
+            .associate { it.identifier to it.controller }
+    )
```

Comment:
Can you add a comment explaining what this is doing?

Reply:
done